neuro_san/internals/run_context/interfaces/tool_caller.py

```diff
@@ -32,9 +32,9 @@ async def make_tool_function_calls(self, component_run: Run) -> Run:
         """
         raise NotImplementedError
 
-    def get_factory(self) -> AgentToolFactory:
+    def get_inspector(self) -> AgentNetworkInspector:
```

Comment:
This reflects the ToolCaller interface change for RunContext land where it needs some object that will answer its questions about the agent network spec.